### PR TITLE
Add repro for `(reg2 == REG_NEXT(reg1)) || (reg2 == REG_LR)` assert in `codegenarm64`.

### PR DESCRIPTION
**Pull request by sandreenko**
This PR adds test for #21363.

The desktop issue is 736188.

These functions need some refactoring before I can push the fix so I want to do it in separate PRs.

### CODE REVIEW

**Comment by sandreenko** [comment]
[Failure example.](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm64_cross_checked_ubuntu16.04_no_tiered_compilation_innerloop_tst_prtest/2029/consoleFull#-906382004f66db340-b3cd-4e23-8190-9b52cb916333)

**Comment by sandreenko** [comment]
PTAL @BruceForstall @dotnet/jit-contrib

File: tests/src/JIT/Regression/JitBlue/DevDiv_736188/DevDiv_736188.csproj [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the configurations -->
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' "></PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' "></PropertyGroup>
+  <PropertyGroup>
+    <DebugType>None</DebugType>
+    <Optimize>True</Optimize>
+  </PropertyGroup>
+  <PropertyGroup>
+    <CLRTestBatchPreCommands><![CDATA[$(CLRTestBatchPreCommands)set COMPlus_JitStressRegs=0x204]]></CLRTestBatchPreCommands>
```
Comment (BruceForstall) [inline tests/src/JIT/Regression/JitBlue/DevDiv_736188/DevDiv_736188.csproj]: Don't you need a newline after `$(CLRTestBatchPreCommands)` and after the complus variable setting?

E.g., one case I found used multiple lines, like:
```
    <CLRTestBatchPreCommands><![CDATA[
$(CLRTestBatchPreCommands)
set COMPlus_DoubleArrayToLargeObjectHeap=0x64
]]></CLRTestBatchPreCommands>
```

Comment (sandreenko) [reply tests/src/JIT/Regression/JitBlue/DevDiv_736188/DevDiv_736188.csproj]: I saw this too, but it have checked that it works without it and looks better in one line.

Comment (sandreenko) [reply tests/src/JIT/Regression/JitBlue/DevDiv_736188/DevDiv_736188.csproj]: Got it, will fix.

**Pull request merged**